src/Microsoft.AspNetCore.SignalR.Client.Core/HubConnection.cs

```diff
@@ -93,7 +93,8 @@ public HubConnection(IConnection connection, IHubProtocol protocol, ILoggerFacto
             using (var memoryStream = new MemoryStream())
             {
                 NegotiationProtocol.WriteMessage(new NegotiationMessage(_protocol.Name), memoryStream);
-                await _connection.SendAsync(memoryStream.ToArray(), _connectionActive.Token);
+                var linkedTcs = CancellationTokenSource.CreateLinkedTokenSource(cancellationToken, _connectionActive.Token);
```

Comment:
We should be checking if the `cancellationToken` passed in is cancel able, and if not avoid the linkedtcs allocations.